File: Signiture-Backend/delivery/db/src/main/java/pcy/study/db/storeuser/StoreUserRepository.java
package pcy.study.db.storeuser;

import org.springframework.data.jpa.repository.JpaRepository;
import pcy.study.db.storeuser.enums.StoreUserStatus;

import java.util.Optional;

public interface StoreUserRepository extends JpaRepository<StoreUser, Long> {

    Optional<StoreUser> findFirstByEmailAndStatusOrderByIdDesc(String email, StoreUserStatus status);
}

File: Signiture-Backend/kotlin/delivery/db/src/main/kotlin/pcy/study/db/storeuser/StoreUserRepository.kt
package pcy.study.db.storeuser

import org.springframework.data.jpa.repository.JpaRepository
import pcy.study.db.storeuser.enums.StoreUserStatus

interface StoreUserRepository : JpaRepository<StoreUser, Long> {

    fun findFirstByEmailAndStatusOrderByIdDesc(
        email: String?,
        status: StoreUserStatus?
    ): StoreUser?
}
